lib/msal-angular/src/msal.interceptor.ts

```diff
@@ -287,25 +273,39 @@ export class MsalInterceptor implements HttpInterceptor {
         absoluteEndpoint
       ).getUrlComponents();
 
-      // Normalized key should include query strings if applicable
-      const relativeNormalizedKey = keyComponents.QueryString
-        ? `${keyComponents.AbsolutePath}?${keyComponents.QueryString}`
-        : this.location.normalize(keyComponents.AbsolutePath);
-
-      // Add resource to matchingResources.relativeResources if same origin, relativeKey matches endpoint, and is not empty
-      if (
-        keyComponents.HostNameAndPort === endpointComponents.HostNameAndPort &&
-        StringUtils.matchPattern(relativeNormalizedKey, absoluteEndpoint) &&
-        relativeNormalizedKey !== "" &&
-        relativeNormalizedKey !== "/*"
-      ) {
-        matchingResources.relativeResources.push(key);
+      // Add resource to matchingResources if keyComponents match endpointComponents
+      if (this.checkUrlComponents(keyComponents, endpointComponents)) {
+        matchingResources.push(key);
       }
     });
 
     return matchingResources;
   }
 
+  /**
+   * Compares URL segments between key and endpoint
+   * @param key 
+   * @param endpoint 
+   * @returns 
+   */
+  private checkUrlComponents(keyComponents: IUri, endpointComponents: IUri): boolean {
```

Comment:
Can we use [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) utility provided by the browser instead of IUri & UrlString helpers? Hoping to remove some of our custom utilities in favor of built-ins

Reply:
Good point! Changed.

Reply:
Thanks!